src/ray/core_worker/experimental_mutable_object_provider.cc

```diff
@@ -22,14 +22,16 @@ namespace experimental {
 
 MutableObjectProvider::MutableObjectProvider(
```

Comment:
Can you also add a C++ unit test to check that ReadRelease is called properly when the shared_ptr goes out of scope?

Reply:
Added.